dipy/io/peaks.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+
+from dipy.core.sphere import Sphere
+from dipy.direction.peaks import (PeaksAndMetrics,
+                                  reshape_peaks_for_visualization)
+from dipy.io.image import save_nifti
+
+
+def load_peaks(fname, verbose=True):
+    """ Load PeaksAndMetrics NPZ file
+    """
+
+    pam_dix = np.load(fname)
+
+    pam = PeaksAndMetrics()
+    pam.affine = pam_dix['affine']
+    pam.peak_dirs = pam_dix['peak_dirs']
+    pam.peak_values = pam_dix['peak_values']
+    pam.peak_indices = pam_dix['peak_indices']
+    pam.shm_coeff = pam_dix['shm_coeff']
+    pam.sphere = Sphere(xyz=pam_dix['sphere_vertices'])
+    pam.B = pam_dix['B']
+    pam.total_weight = pam_dix['total_weight']
+    pam.ang_thr = pam_dix['ang_thr']
+    pam.gfa = pam_dix['gfa']
+    pam.qa = pam_dix['qa']
+    pam.odf = pam_dix['odf']
+
+    if verbose:
+        print('Affine')
+        print(pam.affine)
+        print('Dirs Shape')
+        print(pam.peak_dirs.shape)
+        print('SH Shape')
+        print(pam.shm_coeff.shape)
+        print('ODF')
+        print(pam.odf.shape)
+        print('Total weight')
+        print(pam.total_weight)
+        print('Angular threshold')
+        print(pam.ang_thr)
+        print('Sphere vertices shape')
+        print(pam.sphere.vertices.shape)
+
+
+    return pam
+
+
+def save_peaks(fname, pam, compressed=True):
+    """ Save NPZ file with all important attributes of object PeaksAndMetrics
+    """
+
+    if compressed:
+        save_func = np.savez_compressed
+    else:
+        save_func = np.savez
+
+    save_func(fname,
```

Comment:
Hi @matthieudumont I had one user strongly suggesting to use hdf5 to save the peaks rather than npy or npz. This will allow other non-pythonic projects to read our results. We already depend on pytables that allows to save to hdf5. So, it shouldn't be difficult to make the change. If you agree I can make a PR on top of yours with the required changes. Or if you know how to do it and agree with the suggestion you can go ahead and make the change yourself. See also here https://github.com/nipy/dipy/pull/1147/files
Let me know what you think.